### PR TITLE
demo/non-geo-map

### PR DESCRIPTION
**Pull request by KacperMadej**
Added a cinema seat plan demo.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205567397880579

### CODE REVIEW

**Review by hubertkozik — COMMENTED** [review]
Some inline comments are below and some are here.

- [ ] For a few themes, e.g. BrandDark 'Click to select' label is touching the 'Stage':

image

- [ ] Mobile view is a little bit broken: 
image

- [ ] Exporting to jpg etc. doesn't respect the rotated HTML data labels. I think it can be fixed by using non-HTML data labels with built-in rotation.

File: samples/maps/demo/cinema-seat-plan/demo.details [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+name: Cinema seat plan
+authors:
+  - Kacper Madej
+js_wrap: b
```
[the remark this answers is not in the document]
Comment (KacperMadej) [reply samples/maps/demo/cinema-seat-plan/demo.details:5]: I will add this, although most demos don't have it.

File: samples/maps/demo/cinema-seat-plan/demo.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                String.fromCharCode(65 + Math.floor(i / 11)) + (i % 11 + 1),
+                +taken[i] ? null : void 0,
+                rot[i % 11]
+            ]);
+        }()),
+        keys: ['name', 'value', 'dataLabels.rotation'],
+        joinBy: 'name'
+    }, {
+        // Row & screen labels
+        type: 'mappoint',
```
Comment (marvin19) [inline samples/maps/demo/cinema-seat-plan/demo.js:120]: add `series.accessibility.enabled.false = true` to hide series for row & screen labels from screen readers. Modify `accessibility.point.descriptionFormat` instead

**Review by marvin19 — CHANGES_REQUESTED** [review]

**Review by marvin19 — COMMENTED** [review]
Can you also add:
`accessibility.typeDescription: 'Map of cinema seat plan.'`

**Review by marvin19 — APPROVED** [review]
Looks good!

**Review by TorsteinHonsi — APPROVED** [review]
Thanks, looks good!

File: samples/maps/demo/cinema-seat-plan/demo.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                }
+            }
+        }
+    },
+    tooltip: {
+        outside: true,
+        formatter: function () {
+            return `Seat <b>${this.point.name}</b> ` +
+                (this.point.selected ? 'Selected' : 'Available');
+        }
```
Comment (TorsteinHonsi) [inline samples/maps/demo/cinema-seat-plan/demo.js:67]: We should be able to express this as a string format.

File: samples/maps/demo/cinema-seat-plan/demo.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+                    '00000000110',
+                    '00111000011'
+                ].join('');
+
+            return new Array(66).fill(1).map((seat, i) => [
+                String.fromCharCode(65 + Math.floor(i / 11)) + (i % 11 + 1),
+                +taken[i] ? null : void 0,
+                rot[i % 11]
+            ]);
+        }()),
```
Comment (TorsteinHonsi) [inline samples/maps/demo/cinema-seat-plan/demo.js:125]: This code is quite complex and intimidating. Bearing in mind that the purpose of the demos are to show the public how easy it is to create cool visualizations with Highcharts, I think it is better that we simplify it. Maybe a loop for the rotation is needed, but the array for taken seats I think should be just defined static in the data set.

**Comment by KacperMadej** [comment]
Addressed in https://github.com/highcharts/highcharts/pull/20745

**Pull request merged**